### PR TITLE
Divide Effect Behavior System

### PR DESCRIPTION
**Pull request by cdsupina**
Before this change, each `SpawnableType` had one mega-system to handle all behaviors. I don't this was a good way of handling behaviors because each singular behavior system had to run serially. Making one system for each behavior enables bevy to parallelize the behavior logic.

- Remove effect behavior mega-system
- Add new private systems for each `EffectBehaviorType`
- Add `EffectPlugin` that contains all of the effect related ecs stuff
  - `EffectBehaviorPlugin`
  - `SpawnEffectPlugin`
- Add new `EffectBehaviorType`, `FadeOutAndDespawnAfterAnimation`
  - Despawns effects after the animations is complete while also fading out with an exponential decay
- Fix bug where full animations were not being played (sorry @LordDeatHunter 😅)

### CODE REVIEW

File: src/spawnable/effect/behavior.rs [hunk 1]
```diff
@@ -1,50 +1,211 @@
...
+    FadeOut(Timer),
+    FadeOutAndDespawnAfterAnimation(Stopwatch),
+}
+
+/// Enumerates the types of behaviors that can be performed by effects using
+/// primitive data types for initializing `EffectBehavior` counterparts.
+#[derive(Deserialize, Clone)]
+pub enum EffectBehaviorData {
+    DespawnAfterAnimation,
+    FadeOut(f32),
```
Comment (varoonp123) [inline src/spawnable/effect/behavior.rs:50]: Something say the unit. Maybe a comment. 
```suggestion
    FadeOutMillis(f32),
```

File: assets/sounds/defense_heal.wav
Comment (varoonp123) [inline assets/sounds/defense_heal.wav:1]: Did you mean to delete the audio files?

Comment (cdsupina) [reply assets/sounds/defense_heal.wav:1]: I only meant to merge with main. Aren't we getting the audio files from s3 now?

**Comment by varoonp123** [comment]
`aws --profile shivaji-nonroot s3 cp --dryrun --recursive  s3://assets-thetawave/free_assets/ assets/`

**Review by varoonp123 — APPROVED** [review]
Merge on green

**Pull request merged**